### ISSUE TITLE
Parser should ignore TRUNC() call when parsing TRUNC(CURRENT_DATE) or TRUNC(SYSDATE)

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10668]
The following translation from Oracle to PostgreSQL fails:

```sql
-- Input
select trunc(CURRENT_DATE)-30;

-- Output
select (date_trunc('day', current_date) + -(30));
```

It produces this error:

> [42883]: ERROR: operator does not exist: timestamp with time zone + integer

**Comment by lukaseder** [comment]
The problem here is that in Oracle, `DATE` is really a `TIMESTAMP(0)` type and should be translated as such. If we hard-wire parsing `CURRENT_DATE` to `currentDate().coerce(TIMESTAMP)`, we'll get this output, which is more correct:

```sql
select (date_trunc('day', current_date) + -(30) * interval '1 day');
```

In order to correctly translate all things date time arithmetic from Oracle to X, we probably have to implement a setting that tells the parser to coerce all `DATE` types to `TIMESTAMP(0)`. Perhaps there's a more thorough and configurable solution to mapping types from dialects X to dialects Y. I think I've already created a similar issue but cannot seem to find it.

**Comment by lukaseder** [comment]
The options mentioned previously mostly go quite far. A low hanging fruit here is to create a parsing transformation that removes the `TRUNC()` function call when parsing `TRUNC(CURRENT_DATE)` and `TRUNC(SYSDATE)` in `ORACLE`.

This doesn't solve the problem described above, but works around it. In most cases, retaining this `TRUNC()` expression is undesireable, and we already reproduce it when translating `org.jooq.impl.CurrentDate` to Oracle. I.e. translating

```sql
TRUNC(CURRENT_DATE)-30
```

To Oracle currently yields:

```sql
(TRUNC(TRUNC(SYSDATE), 'DD') - 30)
```

Which is another hint at removing the function call already in the parser being a good idea.

**Comment by lukaseder** [comment]
Again, with this in mind, we could generalise the feature to:

- Specifying a set of parser transformations that could be opted in configuratively
- Turning this into a general SQL transformation that is independent of the parser (more difficult, because we no longer have any input dialect)

**Comment by lukaseder** [comment]
The suggested fix here assumes that `org.jooq.impl.CurrentDate` produces `TRUNC(CURRENT_DATE)`, but it currently produces `TRUNC(SYSDATE)`: https://github.com/jOOQ/jOOQ/issues/9715

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#10967)
